### PR TITLE
Packaging/CI tweaks

### PR DESCRIPTION
**Pull request by nicoddemus**
- Remove `setup.cfg`: wheels are not universal (no Python 2 support), flake8 is not used anymore.
- Add Python 3.9 and Python 3.10 CI.
- Drop "linting" environment, relying on pre-commit.ci instead.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        tox_env: ["py36", "py37", "py38", "linting", "docs"]
```
Comment (prusse-martin) [inline .github/workflows/main.yml:13]: You removed "docs" from tox_env matrix but didn't remove the "testenv:docs" from tox.ini file.
Is that intentional?

**Review by prusse-martin — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (prusse-martin) [reply .github/workflows/main.yml:13]: We should add a notification e-mail in readthedocs to better notice failures in building the docs.
https://readthedocs.org/dashboard/conda-devenv/notifications/

Comment (nicoddemus) [reply .github/workflows/main.yml:13]: > You removed "docs" from tox_env matrix but didn't remove the "testenv:docs" from tox.ini file.
Is that intentional?

Yes, wanted to leave a way to easily build the docs locally.

> We should add a notification e-mail in readthedocs to better notice failures in building the docs.

You think that's necessary? Note that now we have a dedicated ReadTheDocs build:

image

If you prefer to have the email notification: which email should we use?

Comment (prusse-martin) [reply .github/workflows/main.yml:13]: I did ask myself that =)
I will be nice if git hub have one that just forwards to the maintainers but ask far I can tell there is not one.
Well I will add my.

Comment (nicoddemus) [reply .github/workflows/main.yml:13]: OK 👍

**Pull request merged**